equinox_monorepo/neural_cde/neural_cde.py

```diff
@@ -0,0 +1,190 @@
+        - `activation`: The activation between each hidden layer. Defaults to GeLU.
+        - `final_activation`: The activation function after the output layer. Defaults to
+            tanh, this is a good choice for neural CDEs.
+        - `n_classes`: The number of output classes.
+        - `interpolator`: The Diffrax interpolator class to use for interpolation.
+            Defaults to `diffrax.LinearInterpolation.`
+        - `solver`: The Diffrax solver to use to solve the neural CDE. Defaults to
+            `diffrax.Tsit5()`.
+        - `mode`: Determines how to go from an output of shape (times, feats) to
+            (feats) for use in classification. Uses the enum `Mode`. Defaults to
+            `Mode.terminal`.
+        - `key`: A `jax.random.PRNGKey` used to provide randomness for parameter
+            initialisation.
+        """
+        super().__init__(**kwargs)
+        encoder_key, mlp_key, decoder_key = jr.split(key, 3)
+        self.encoder = eqx.nn.MLP(
+            data_size, model_size, encoder_width, encoder_depth, key=encoder_key
+        )
+        self.vector_field = VectorField(
+            data_size,
+            model_size,
+            mlp_hidden_width,
+            mlp_depth,
+            activation,
+            final_activation,
+            key=mlp_key,
+        )
+        # If there are 2 output classes, we only need a 1-dimensional output for
+        # classification.
+        if n_classes == 2:
+            out_dim = 1
+        else:
+            out_dim = n_classes
+
+        self.decoder = eqx.nn.Linear(model_size, out_dim, key=decoder_key)
+        self.interpolator = interpolator
+        self.solver = solver
+        self.mode = mode
+
+    def __call__(self, times: Float[Array, "times"], xs: Float[Array, "times feats"]):
+        if self.interpolator == dfx.CubicInterpolation:
+            ys = dfx.backward_hermite_coefficients(times, xs)
+        else:
+            ys = xs
+        control = self.interpolator(times, ys)
+        cde_term = dfx.ControlTerm(self.vector_field, control)
+        ode_term = cde_term.to_ode()
+        dt0 = None
+        y0 = self.encoder(control.evaluate(times[0]))
+        saveat = dfx.SaveAt(t1=True)
+        solution = dfx.diffeqsolve(
+            ode_term,
+            self.solver,
+            times[0],
+            times[-1],
+            dt0,
+            y0,
+            stepsize_controller=dfx.PIDController(rtol=1e-3, atol=1e-6),
+            saveat=saveat,  # pyright: ignore
+        )
+
+        if self.mode == Mode.terminal:
+            out = solution.ys[-1]
+        elif self.mode == Mode.meanpool:
+            out = jnp.mean(solution.ys, axis=0)
+        else:
+            raise ValueError(
+                "'mode' should be either `Mode.terminal` or "
+                f"`Mode.meanpool` using the `Mode` enum type. Got {self.mode} "
+            )
+
+        return jnn.softmax(self.decoder(out))
```

Comment:
If self.decoder has output dimension of 1 I think using softmax as the final activation will fix the output to always be 1, could either have binary classification's changed to have output dimension of 2 or use a different final activation when n_classes=2